contracts/Collection.sol

```diff
@@ -0,0 +1,123 @@
+pragma solidity ^0.8.0;
+
+import "./freeportParts/BaseNFT.sol";
+import "./Freeport.sol";
+
+/** This this contract describes the collection of NFTs associated with a particular user.
+ *
+ */
+contract Collection is BaseNFT {
+    function initialize() public initializer {
+        __BaseNFT_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        owner = _msgSender();
+    }
+
+    // Name of the collection for open sea.
+    string public name;
+    // Owner contract address.
+    address public owner;
+    // Royalty manager role.
+    bytes32 public constant ROYALTY_MANAGER_ROLE = keccak256("ROYALTY_MANAGER_ROLE");
+    // The address of Freeport contract.
+    Freeport public freeport;
+    /** A counter of NFT types issued.
+     * This is used to generate unique NFT IDs.
+     */
+    uint32 public idCounter;
+
+    function setName(string memory newName) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        name = newName;
+    }
+
+    function setRoyaltyManager(address royaltyManager) public {
```

Comment:
There is already a function `grantRole`. It’s enough to just explain the role in the documentation of the contract (/** … */ above).